From 4484707e86299505a98e7b19bef5f61d3aa98ea3 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:19:53 -0400
Subject: [fs] nfs: use ARRAY_SIZE instead of sizeof/sizeof[0]

Message-id: <1411050057-12417-31-git-send-email-steved@redhat.com>
Patchwork-id: 96194
O-Subject: [RHEL7.1 PATCH 30/94] NFS: use ARRAY_SIZE instead of sizeof/sizeof[0]
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Fabian Frederick <fabf@skynet.be>

Use macro definition

Cc: Trond Myklebust <trond.myklebust@primarydata.com>
Cc: linux-nfs@vger.kernel.org
Signed-off-by: Fabian Frederick <fabf@skynet.be>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: 002160269fcc73a01ca3889d3011afc9b63a53bd
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/super.c b/fs/nfs/super.c
index e08eba9..bb7762d 100644
--- a/fs/nfs/super.c
+++ b/fs/nfs/super.c
@@ -1026,8 +1026,7 @@ static bool nfs_auth_info_add(struct nfs_auth_info *auth_info,
          rpc_authflavor_t flavor)
 {
  unsigned int i;
- unsigned int max_flavor_len = (sizeof(auth_info->flavors) /
-           sizeof(auth_info->flavors[0]));
+ unsigned int max_flavor_len = ARRAY_SIZE(auth_info->flavors);
 
  /* make sure this flavor isn't already in the list */
  for (i = 0; i < auth_info->flavor_len; i++) {
-- 
1.7.1